### DISCUSSION TITLE
The new OpenSceneGraph website

### DISCUSSION THREAD

**Discussion by AnyOldName3** [openscenegraph/OpenSceneGraph#1292] [Show and tell]
As mentioned in Robert's announcement [here](https://github.com/openscenegraph/OpenSceneGraph/discussions/1291), I've been tasked with replacing the OSG's old Joomla-based site with a new one. There were several reasons for this including the cost of running the old site, it not matching the VSG site or modern tastes, and it being a pain to keep it maintained and secure.

### Me

For those who don't know who I am, and care for whatever reason, I've been using the OpenSceneGraph since 2017 as a graphics developer for [OpenMW](https://openmw.org/). I'm not usually a web developer, but I'm not totally alien to it, either, and at the minimum, know how much of a pain it can be when a software library has a bad website, or throws away all its old documentation when a new product is launched to replace it.

### Archive

For continuity of service, and to avoid contributing to link rot or throwing away the corpus of knowledge contained within the mailing list archive, I've started by creating a static archive of everything that still works on the old site. For now, you can get at a replica of www.openscenegraph.com [here](https://openscenegraph.github.io/OpenSceneGraphDotComBackup/OpenSceneGraph/www.openscenegraph.com/index.html) and the `osg-users` mailing list archive [here](https://openscenegraph.github.io/OpenSceneGraphDotComBackup/OpenSceneGraph/lists.openscenegraph.org/pipermail/osg-users-openscenegraph.org/index.html). Don't bookmark them just yet as the URLs will likely change. There are a few things that inherently won't work thanks to these being static, like logging in or using the mailing list's search tool, but otherwise, I'd rather have as much still function as possible. If you find any problems, feel free to report them [here](https://github.com/openscenegraph/OpenSceneGraphDotComBackup/issues). Bear in mind, though, that lots of things already aren't working on the live site, so make sure to double-check the problem is with the archive rather than the problem's been accurately archived.

To try and make things as seamless as possible, I'm aiming to have a system to convert links to the old site into links to the archive. If it can be done cheaply, this might end up running on openscenegraph.com, and offer a link to the new site, too, but plans are far from finalised. If I do this right, then archive results should continue showing up in search engines whenever the old site would, which is an added bonus.

### The New Site

Now onto the exciting part - the new site. Having an archive of its predecessor means there's freedom to streamline it and discard outdated information from the old site without ruining anyone's day. To that end, I've made it from scratch with Jekyll (just like the VSG site). This is now live at [openscenegraph.github.io/openscenegraph.io](https://openscenegraph.github.io/openscenegraph.io/).

As you might imagine, the purpose of the new site is less about trying to sell people on using the OpenSceneGraph, and more on making sure people know it exists, but should be using the VulkanSceneGraph for new projects instead, while still pointing existing users to where they can find the information they need.

A few differences you might notice include:
* There's no longer a news/announcements section. As the OpenSceneGraph isn't expecting any big news, this is no longer necessary, and if it turns out we're wrong about this, it'll be easy to bring back.
* The guides and knowledge base are gone. If you're still using these, there's no need to worry, as they're in the old site archive. However, most of what I looked through was in a pretty dire state, and would have needed a rewrite to avoid being misleading. That's not obviously a worthwhile endeavour as we're expecting new projects to use the VSG, and existing OSG users to have more of an idea how to use it. A benefit of the new site is that it should be easier for community members to submit a new guide via a pull request than it was to use the Joomla site, though, so if anyone strongly disagrees, they can help fix it.
* The downloads section is also gone. This didn't actually host any downloads other than source zips which are also available on GitHub, although there were links to (mostly outdated) third-party prebuilt files. We might end up listing those somewhere if necessary.
* A bunch of community-related pages that haven't been updated for years, like the event listings, are also gone.
* The new site behaves itself better on particularly large or small displays, and when the device pixel ratio isn't exactly one (think retina displays etc.).
* It'll seem more polished. The old site had a surprising number of problems like elements overflowing their boundaries and overlapping each other.
* It has some newer, prettier screenshots, in particular for OpenMW. When migrating existing use case screenshots, I noticed that many were really old and low-resolution or otherwise generally unflattering. I've put some effort into making adding screenshots and having them behave nicely (e.g. serving up an appropriate-resolution thumbnail for the current display) more straightforward than it might be on a typical Jekyll site. I'm expecting that this system will make its way to the VSG site, too, as that's currently not got any information about existing downstream projects.

### Next steps for the community

* Try out the archive and make sure the information you need is still accessible.
* Try out the new site and give me your feedback.
* If your application's missing from the Screenshots page, or just has old screenshots from the old site that you'd rather revamp, consider submitting some new ones. In particular, if you've got a VulkanSceneGraph port, that would be great to mention as well. There are instructions for this [here](https://openscenegraph.github.io/openscenegraph.io/gallery/readme.html).

**Comment by trajcenikolovnick** [comment]
Hi Chris, (all)

I have some cool videos and images from some projects I would like to
upload. I went through the instructions, however I am completely dumb for
this sort of IT :-/ .. Anyone (You?) can and willing to help? I have posted
them on my LinkedIn profile https://www.linkedin.com/in/tnick ... I really
like the idea about posting images and videos of the old (and great!) OSG -
been with it for more then 2 decades. All the videos and screenshots are
from OSG projects .. Let me know and thanks a bunch!

Cheers,
Nick

Al



On Wed, Jan 10, 2024 at 8:00 PM Chris Djali ***@***.***>
wrote:

> As mentioned in Robert's announcement here
> <https://github.com/openscenegraph/OpenSceneGraph/discussions/1291>, I've
> been tasked with replacing the OSG's old Joomla-based site with a new one.
> There were several reasons for this including the cost of running the old
> site, it not matching the VSG site or modern tastes, and it being a pain to
> keep it maintained and secure.
> Me
>
> For those who don't know who I am, and care for whatever reason, I've been
> using the OpenSceneGraph since 2017 as a graphics developer for OpenMW
> <https://openmw.org/>. I'm not usually a web developer, but I'm not
> totally alien to it, either, and at the minimum, know how much of a pain it
> can be when a software library has a bad website, or throws away all its
> old documentation when a new product is launched to replace it.
> Archive
>
> For continuity of service, and to avoid contributing to link rot or
> throwing away the corpus of knowledge contained within the mailing list
> archive, I've started by creating a static archive of everything that still
> works on the old site. For now, you can get at a replica of
> www.openscenegraph.com here
> <https://openscenegraph.github.io/OpenSceneGraphDotComBackup/OpenSceneGraph/www.openscenegraph.com/index.html>
> and the osg-users mailing list archive here
> <https://openscenegraph.github.io/OpenSceneGraphDotComBackup/OpenSceneGraph/lists.openscenegraph.org/pipermail/osg-users-openscenegraph.org/index.html>.
> Don't bookmark them just yet as the URLs will likely change. There are a
> few things that inherently won't work thanks to these being static, like
> logging in or using the mailing list's search tool, but otherwise, I'd
> rather have as much still function as possible. If you find any problems,
> feel free to report them here
> <https://github.com/openscenegraph/OpenSceneGraphDotComBackup/issues>.
> Bear in mind, though, that lots of things already aren't working on the
> live site, so make sure to double-check the problem is with the archive
> rather than the problem's been accurately archived.
>
> To try and make things as seamless as possible, I'm aiming to have a
> system to convert links to the old site into links to the archive. If it
> can be done cheaply, this might end up running on openscenegraph.com, and
> offer a link to the new site, too, but plans are far from finalised. If I
> do this right, then archive results should continue showing up in search
> engines whenever the old site would, which is an added bonus.
> The New Site
>
> Now onto the exciting part - the new site. Having an archive of its
> predecessor means there's freedom to streamline it and discard outdated
> information from the old site without ruining anyone's day. To that end,
> I've made it from scratch with Jekyll (just like the VSG site). This is now
> live at openscenegraph.github.io/openscenegraph.io.
>
> As you might imagine, the purpose of the new site is less about trying to
> sell people on using the OpenSceneGraph, and more on making sure people
> know it exists, but should be using the VulkanSceneGraph for new projects
> instead, while still pointing existing users to where they can find the
> information they need.
>
> A few differences you might notice include:
>
>    - There's no longer a news/announcements section. As the
>    OpenSceneGraph isn't expecting any big news, this is no longer necessary,
>    and if it turns out we're wrong about this, it'll be easy to bring back.
>    - The guides and knowledge base are gone. If you're still using these,
>    there's no need to worry, as they're in the old site archive. However, most
>    of what I looked through was in a pretty dire state, and would have needed
>    a rewrite to avoid being misleading. That's not obviously a worthwhile
>    endeavour as we're expecting new projects to use the VSG, and existing OSG
>    users to have more of an idea how to use it. A benefit of the new site is
>    that it should be easier for community members to submit a new guide via a
>    pull request than it was to use the Joomla site, though, so if anyone
>    strongly disagrees, they can help fix it.
>    - The downloads section is also gone. This didn't actually host any
>    downloads other than source zips which are also available on GitHub,
>    although there were links to (mostly outdated) third-party prebuilt files.
>    We might end up listing those somewhere if necessary.
>    - A bunch of community-related pages that haven't been updated for
>    years, like the event listings, are also gone.
>    - The new site behaves itself better on particularly large or small
>    displays, and when the device pixel ratio isn't exactly one (think retina
>    displays etc.).
>    - It'll seem more polished. The old site had a surprising number of
>    problems like elements overflowing their boundaries and overlapping each
>    other.
>    - It has some newer, prettier screenshots, in particular for OpenMW.
>    When migrating existing use case screenshots, I noticed that many were
>    really old and low-resolution or otherwise generally unflattering. I've put
>    some effort into making adding screenshots and having them behave nicely
>    (e.g. serving up an appropriate-resolution thumbnail for the current
>    display) more straightforward than it might be on a typical Jekyll site.
>    I'm expecting that this system will make its way to the VSG site, too, as
>    that's currently not got any information about existing downstream projects.
>
> Next steps for the community
>
>    - Try out the archive and make sure the information you need is still
>    accessible.
>    - Try out the new site and give me your feedback.
>    - If your application's missing from the Screenshots page, or just has
>    old screenshots from the old site that you'd rather revamp, consider
>    submitting some new ones. In particular, if you've got a VulkanSceneGraph
>    port, that would be great to mention as well. There are instructions for
>    this here
>    <https://openscenegraph.github.io/openscenegraph.io/gallery/readme.html>
>    .
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openscenegraph/OpenSceneGraph/discussions/1292>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC7ND7DOP3UUTO4YJPZNI43YN3JK5AVCNFSM6AAAAABBVGUGD6VHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZWGA2TMOJVG4>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
trajce nikolov nick

**Reply by AnyOldName3** [reply]
Hi Nick,

First things first, sorry for the delay - I've been away for a few days, so am only just now getting a chance to get back to OSG/VSG stuff.

If you've got some images/videos and text about a project, e.g. OpenIG, provided it's not everyone asking me to do it for loads of projects, it should be fine just to email them to me, but I can also work on clarifying the instructions if I know where people are hitting problems. It's probably best not to use LinkedIn as an intermediary, though, as it rescales and recompresses images and doesn't let you download videos. Also, on your LinkedIn, I only see three directly-posted images or videos, and then some of your reposted content is Unreal Engine and Unity stuff, so it's not clear what I'd want to collect without some input from your end, anyway.

Cheers,

Chris

**Comment by trajcenikolovnick** [comment]
Hi Chris !

Thanks a bunch ! Let me send you on email some of the content I have in
mind. The videos are available on youtube. From the whole LI gallery there
is only one screenshot of Diligent (the Digital Twin content was generated
by an OSG app), and no UE or Unity at all ... Anyhow, I will contact you.

Thanks once again!

Cheers,
Nick

On Tue, Feb 6, 2024 at 8:03 PM Chris Djali ***@***.***> wrote:

> Hi Nick,
>
> First things first, sorry for the delay - I've been away for a few days,
> so am only just now getting a chance to get back to OSG/VSG stuff.
>
> If you've got some images/videos and text about a project, e.g. OpenIG,
> provided it's not everyone asking me to do it for loads of projects, it
> should be fine just to email them to me, but I can also work on clarifying
> the instructions if I know where people are hitting problems. It's probably
> best not to use LinkedIn as an intermediary, though, as it rescales and
> recompresses images and doesn't let you download videos. Also, on your
> LinkedIn, I only see three directly-posted images or videos, and then some
> of your reposted content is Unreal Engine and Unity stuff, so it's not
> clear what I'd want to collect without some input from your end, anyway.
>
> Cheers,
>
> Chris
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openscenegraph/OpenSceneGraph/discussions/1292#discussioncomment-8386068>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC7ND7A53SEJJVUFBMJGWEDYSJV6ZAVCNFSM6AAAAABBVGUGD6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DGOBWGA3DQ>
> .
> You are receiving this because you commented.Message ID:
> <openscenegraph/OpenSceneGraph/repo-discussions/1292/comments/8386068@
> github.com>
>


-- 
trajce nikolov nick

**Comment by AnyOldName3** [comment]
The old site is now gone. The link redirector service is live, so if you've got bookmarks to the old site, or links to specific pages on other sites, they'll still work, but go to the archive instead. I'd still recommend you switch them over if you can, otherwise if the redirector goes down or gets removed, they'll stop working.
